valhalla/sif/dynamiccost.h

```diff
@@ -828,6 +840,8 @@ class DynamicCost {
   // and bicycle generally allow access (with small penalties).
   bool allow_destination_only_;
 
+  bool allow_conditional_destination_restrictions_;
```

Comment:
I did that for fine control, now we can disallow one of them separately. I can replace if we want to mix them together.

Reply:
I can't come up with a case when we need them separate

Reply:
For example, we can to relax gradually. A path can be found after relaxing by one parameter and no need allow others.
But as you say, done

Reply:
Rolled back to the first version, because `allow_destination_only_` is true by default.

Reply:
I think by default `allow_destination_only_ = true` because of pedestrian and bicycle profiles. We set to `false` only for the first pass of bidirectional astar https://github.com/valhalla/valhalla/blob/master/src/thor/route_action.cc#L370